examples/stream-hello-world/StreamHelloWorld_Client.cpp

```diff
@@ -26,20 +28,22 @@ int main(int argc, char* argv[]) {
   {
     // this example runs inside the Future.then lambda
     LOG(INFO) << "------------------ Run in future.then";
-    auto s = std::make_shared<ExampleSubscriber>(5, 6);
+    auto s = yarpl::Reference<yarpl::Subscriber<Payload>>(
+        new ExampleSubscriber(5, 6));
     rsf->connect().then([s](std::shared_ptr<RSocketRequester> rs) {
-      rs->requestStream(Payload("Bob"), s);
+      rs->requestStream(Payload("Bob"))->subscribe(s);
     });
-    s->awaitTerminalEvent();
+    //    s->awaitTerminalEvent();
```

Comment:
Specifically, I have to do this:

```cpp
auto s = yarpl::Reference<yarpl::Subscriber<Payload>>(new ExampleSubscriber(5, 6));
```

but I want this:

```cpp
auto s = yarpl::Reference<ExampleSubscriber>(new ExampleSubscriber(5, 6));
```

so that I can call methods of `ExampleSubscriber`.

But that can't be passed into `subscribe`:

> error: no viable conversion from 'Reference<rsocket_example::ExampleSubscriber>' to 'Reference<Subscriber<reactivesocket::Payload>>'